packages/teuchos/numerics/src/Teuchos_SerialDenseMatrix.hpp

```diff
@@ -612,6 +618,40 @@ int SerialDenseMatrix<OrdinalType, ScalarType>::putScalar( const ScalarType valu
   return 0;
 }
 
+template<typename OrdinalType, typename ScalarType> void
+SerialDenseMatrix<OrdinalType, ScalarType>::swap(
+  SerialDenseMatrix<OrdinalType, ScalarType> &B)
+{
+  // Notes:
+  // > DefaultBLASImpl::SWAP() uses a deep copy. This fn uses a pointer swap.
+  // > this fn covers both Vector and Matrix, such that some care must be
+  //   employed to not swap across types (creating a Vector with non-unitary
+  //   numCols_)
+  // > Inherited data that is not currently swapped (since inactive/deprecated):
+  //   >> Teuchos::CompObject:
+  //        Flops *flopCounter_ [Note: all SerialDenseMatrix ctors initialize a
+  //        NULL flop-counter using CompObject(), such that any flop increments
+  //        that are computed are not accumulated.]
+  //   >> Teuchos::Object: (now removed from inheritance list)
+  //        static int tracebackMode (no swap for statics)
+  //        std::string label_ (has been reported as a cause of memory overhead)
+
+  // cache B values
+  ScalarType* B_ptr  = B.values_;
```

Comment:
Just use `std::swap`, instead of making temporaries.

https://en.cppreference.com/w/cpp/algorithm/swap

Reply:
I think you are suggesting using std:swap() on the low-level attribute data (pointer, boolean, and size), correct?  That would be fine and less code.

Reply:
@mseldre Yes, please :-)

Reply:
This does introduce a dependence on <algorithm> in the Teuchos headers.

Reply:
Hmm, dropped my header name due to my decoration -- try that again:

Note that this does introduce a dependence on algorithm.hpp in the Teuchos headers

Reply:
Ok done -- tested and committed within Dakota.  @rhoope : I'll try to figure out how to update the PR code, but feel free to shorten my search if it's non-obvious.  ;)

Reply:
I'm setup to update the PR branch and will incorporate your Dakota changes into the PR.

Reply:
@rhoope --  Looks like I need you to stage an additional Trilinos commit, linked to this PR.  Can you do this or show me how?  Patch has been committed on devel branch for dakota/packages/external.

@mhoemmen -- You might also consider whether Flops *flopCounter_ is important to swap.  It looked like an afterthought to me and not worth proliferating, but you would know better whether this is important.

Reply:
> algorithm.hpp

Use `<utility>` since Teuchos assumes C++11 these days.

Reply:
I didn't need to include anything additional as the dependency was already resolved.  @rhoope presumably did for the unit tests, as he mentions below.  

Any concerns on flopCounter_ before moving forward?